specification/maps/data-plane/Search/preview/2022-09-01-preview/geocoding.json

```diff
@@ -0,0 +1,1001 @@
+          }
+        }
+      }
+    },
+    "/reverseGeocode:batch": {
+      "post": {
+        "description": "**Reverse Geocoding Batch API**\n\n\n**Applies to**: see pricing [tiers](https://aka.ms/AzureMapsPricingTier).\n\n\n\nThe Reverse Geocoding Batch API sends batches of queries to [Reverse Geocoding API](https://docs.microsoft.com/en-us/rest/api/maps/search-v2/get-reverse-geocoding) using just a single API call. The API allows caller to batch up to **100** queries.\n>[!Important]\n>By using this feature, you agree to the preview legal terms. See the [Preview Supplemental Terms](https://azure.microsoft.com/en-us/support/legal/preview-supplemental-terms/) for additional details.\n\n### Submit Synchronous Batch Request\nThe Synchronous API is recommended for lightweight batch requests. When the service receives a request, it will respond as soon as the batch items are calculated and there will be no possibility to retrieve the results later. The Synchronous API will return a timeout error (a 408 response) if the request takes longer than 60 seconds. The number of batch items is limited to **100** for this API.\n```\nPOST https://atlas.microsoft.com/reverseGeocode/batch?api-version=2022-09-01-preview\n```\n### POST Body for Batch Request\nTo send the _reverse geocoding_ queries you will use a `POST` request where the request body will contain the `batchItems` array in `json` format and the `Content-Type` header will be set to `application/json`. Here's a sample request body containing 2 _reverse geocoding_ queries:\n\n\n```\n{\n  \"batchItems\": [\n    {\n      \"coordinates\": [-122.128275, 47.639429],\n      \"resultTypes\": [\"Address\", \"PopulatedPlace\"]\n    },\n    {\n      \"coordinates\": [-122.341979399674, 47.6095253501216]\n    }\n  ]\n}\n```\n\nA _reverse geocoding_ batchItem object can accept any of the supported _reverse geocoding_ [URI parameters](https://docs.microsoft.com/en-us/rest/api/maps/search-v2/get-reverse-geocoding#uri-parameters) except query.\n\n\nThe batch should contain at least **1** query.\n\n\n### Batch Response Model\nThe batch response contains a `summary` component that indicates the `totalRequests` that were part of the original batch request and `successfulRequests` i.e. queries which were executed successfully. The batch response also includes a `batchItems` array which contains a response for each and every query in the batch request. The `batchItems` will contain the results in the exact same order the original queries were sent in the batch request. Each item is of one of the following types:\n\n  - [`GeocodingResponse`](https://docs.microsoft.com/en-us/rest/api/maps/search-v2/get-reverse-geocoding#geocodingresponse) - If the query completed successfully.\n\n  - `Error` - If the query failed. The response will contain a `code` and a `message` in this case.\n\n\n",
+        "operationId": "Search_PostReverseGeocodingBatch",
```

Comment:
Another case of using the HTTP verb as part of the method name.

Reply:
Thanks for the remind here.
This is the format we used for all Azure Maps APIs documentation, and it also won't cause any issue in SDK as well.

Reply:
@JeffreyRichter do we have official guidance here?  IMO using the verb as the prefix of the API name is a big mistake.

Reply:
Azure Maps was originally not an Azure service and they are an older service created before guidelines existed. We do have some guidance on operation naming (like use CreateOrReplace for PUT and CreateOrUpdate for PATCH) but we'd put this operation in the POST-action category and the operation ID should be the verb action so probably should be "Search_ReverseGeocodingBatch" (without the Post). But there should be an action verb; maybe 'Query' since I see this in the documentation several times.: "Search_QueryReverseGeocoding"? Also "Batch" is probably not necessary in the Operation ID.

Reply:
There are also other teams using this format as operationId.
It will be a big breaking change for our documentation if I remove http method from the operationId since it will cause the format of the titles in the documentation be inconsistent.

Example of other API using this format:
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v3.0/LUIS-Runtime.json
https://learn.microsoft.com/en-us/rest/api/language/2022-10-01-preview/question-answering-projects/get-project-details?tabs=HTTP

Reply:
IMO we shouldn't compromise on an API name to make it match the docs, nor should bad actors be used as a justification.  Once the API has been released, we can't change the name and will live with it for years, so we should get it right the first time.

I get that for REST docs it makes sense to use REST nomenclature.  However, this is an implementation detail, and while it's highly unlikely to change, we shouldn't be leaking that in the API names where we try to be language/implementation agnostic.

Reply:
@JeffreyRichter 
About the value of "operationId", could you share the existing related API guidance as a clear reference? And I don't understand the difference between verb action and action verb.

Reply:
@jhendrixMSFT 
For the comment, your original concern is that current operationId value would introduce HTTP verb in SDK method name. After confirming with SDK members, the case won't happen on SDK side. Then, you introduce another concern about the HTTP verb in operationId. What's the API name you refer to in official Azure API document website? https://learn.microsoft.com/en-us/rest/api/cognitiveservices-luis/runtimev3.0/prediction/get-slot-prediction-get?tabs=HTTP

Reply:
I'm referring to the SDKs for various languages (C#, Python, etc).

The value of the `operationId` field is used to generate type and method names in the various SDKs.  The value to the left of the `_` is for the type, and to the right is the method.  In this case, you'd have a type named `Service` (this would be the service client name) with a method named `PostGeocodingBatch`.

Some language code generators will use these values verbatim and some won't.  It's best for the operation IDs to match the desired names of the types/methods to be generated in a language-agnostic way.

Reply:
> @JeffreyRichter About the value of "operationId", could you share the existing related API guidance as a clear reference? And I don't understand the difference between verb action and action verb.

https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#operation-and-operationid

Reply:
Hi @jhendrixMSFT, @JeffreyRichter ,
Really appreciate for the clear explanation.
I work with our SDK team very closely, and I have discussed this issue with them recently.
They told me they still need to change the function name when working on the SDK, so this design won't cause any issue there.
Hope this information can remove your worries and concerns.
Please let me know how you think.
Thanks a lot for your help!

Reply:
@john35452 It may be common to override operationId but we want to avoid it wherever possible. What is the method name they want to override it to?

Reply:
Hi @mikekistler ,

Thanks for the reply here!
I would like to give more details about how we use this operationId.
As we discussed, the operationId will be used to generate the method name, and we will use convenience layer to override the operationId afterward.
However, we also use this operationId to generate the title of the documentation title.
This logic is implemented in our pipeline, so I cannot change this.
If I change the operationId here, the titles of APIs in the document will be inconsistent.
So, the reason why I would like to keep this operationId is not to make it consistent with documents but to make the documents consistent.
Further, I also think it might not be suitable for me to change or break the whole document.
For the new method name in SDK, I believe it's not the most critical problem for this issue, because we still use operationId to generate documentation title.

Really appreciate for your reading and understanding!
Please also let me know how you think about this.
cc @JeffreyRichter @jhendrixMSFT

Reply:
@john35452 You say:
> If I change the operationId here, the titles of APIs in the document will be inconsistent.

Inconsistent with what?

If you change the operationId so that it can be used to directly create the SDK method names, that means that REST API documentation and SDKs _will be_ consistent -- and that is a good thing.

Reply:
Hi @mikekistler ,
Thanks for the input here!
I have discussed within our team, and I think we can use an action in the operationId rather than a http method to make the naming more reasonable.
Really appreciate for your help!
https://github.com/Azure/azure-rest-api-specs/pull/19951/commits/1dda592e388b93ace03a63945cd38f5bc1fe5328

cc @JeffreyRichter @jhendrixMSFT @evaou